torchvision/transforms/v2/_augment.py

```diff
@@ -135,3 +138,150 @@ def _transform(
             inpt = F.erase(inpt, **params, inplace=self.inplace)
 
         return inpt
+
+
+class _BaseMixupCutmix(Transform):
+    def __init__(self, *, alpha: float, num_categories: Optional[int] = None, labels_getter="default") -> None:
+        super().__init__()
+        self.alpha = alpha
+        self._dist = torch.distributions.Beta(torch.tensor([alpha]), torch.tensor([alpha]))
+
+        self.num_categories = num_categories
+
+        self.labels_getter = labels_getter
+        self._labels_getter = _parse_labels_getter(labels_getter)
+
+    def forward(self, *inputs):
+        inputs = inputs if len(inputs) > 1 else inputs[0]
+        flat_inputs, spec = tree_flatten(inputs)
+        needs_transform_list = self._needs_transform_list(flat_inputs)
+
+        if has_any(flat_inputs, PIL.Image.Image, datapoints.BoundingBox, datapoints.Mask):
+            raise TypeError(f"{type(self).__name__}() does not support PIL images, bounding boxes and masks.")
+
+        labels = self._labels_getter(inputs)
+        if labels is None:
+            msg = "Couldn't find a label in the inputs."
+            if self.labels_getter == "default":
+                msg = f"{msg} To overwrite the default find behavior, pass a callable for labels_getter."
+            raise RuntimeError(msg)
+        elif not isinstance(labels, torch.Tensor):
+            raise ValueError(f"The labels must be a torch.Tensor, but got {type(labels)} instead.")
+        elif labels.ndim in {1, 2}:
+            if labels.ndim == 2 and self.num_categories is not None and labels.shape[-1] != self.num_categories:
+                raise ValueError(
+                    f"2D labels are assumed to be probability based, "
+                    f"but the number of elements in last dimension does not match the number of categories: "
+                    f"{labels.shape[-1]} != {self.num_categories}."
+                )
+        else:
+            raise ValueError(
+                f"labels should be a index based with shape (batch_size,) "
+                f"or a probability based with shape (batch_size, num_categories), "
+                f"but got a tensor of shape {labels.shape} instead."
+            )
+
+        params = {
+            "labels": labels,
+            "batch_size": labels.shape[0],
+            **self._get_params(
+                [inpt for (inpt, needs_transform) in zip(flat_inputs, needs_transform_list) if needs_transform]
+            ),
+        }
+
+        # By default, the labels will be False inside needs_transform_list, since they are a torch.Tensor, but coming
+        # after an image or video. However, since we want to handle them in _transform, we
+        needs_transform_list[next(idx for idx, inpt in enumerate(flat_inputs) if inpt is labels)] = True
```

Comment:
We used a different strategy in `SanitizeBoundingBox` where we called `_transform()` on all inputs and just handled that filtering logic within `_transform()`. I don't have a pref right now (haven't thought about it much). But maybe we should align both transforms to follow the same strat? (we could do it in another PR)

Reply:
We can't use the same strategy here. `SanitizeBoundingBox` does not affect images or videos, so we don't care about `needs_transform_list` there. However, here we transform images. Meaning, we need to use `needs_transform_list` to make use of the heuristic about what image to transform. This cannot be done in `_transform` since in there we have no concept if an image should be transformed or not.

Reply:
Shouldn't we transform all images ?

Reply:
I think I understand what you mean: we'd need to re-implement the "tensor pass-through heuristic" in `_transform()` if we were to do something like in `SanitizeBoundingBox()`, and we don't want to do that. I feel like we could use the same strategy used here in `SanitizeBoudingBox()` though. But that's OK.

> Shouldn't we transform all images (each image is collated as (N, C, H, W)) ?

We are transforming all images yes

Reply:
> I feel like we could use the same strategy used here in `SanitizeBoudingBox()` though. But that's OK.

Yes, we can certainly also use `needs_transform_list` there. I'm ok with that. Up to you.